### PR TITLE
Move overview group and filter state to redux

### PR DESCRIPTION
**Pull request by TheRealJon**
Fixes https://github.com/openshift/console/issues/1296

Moved state related to overview grouping and filtering into redux.

### CODE REVIEW

File: frontend/public/ui/ui-reducers.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { types } from './ui-actions';
 import { ALL_NAMESPACES_KEY, LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY, NAMESPACE_LOCAL_STORAGE_KEY } from '../const';
 import { AlertStates, isSilenced, SilenceStates } from '../monitoring';
 import { legalNamePattern, getNamespace } from '../components/utils/link';
+import { GROUP_BY_APPLICATION } from '../components/overview';
```
Comment (spadgett) [inline frontend/public/ui/ui-reducers.js]: This import will pull overview into the main bundle

Comment (TheRealJon) [reply frontend/public/ui/ui-reducers.js]: Moved the shared constants into a separate `constants.ts` file

**Comment by spadgett** [comment]
If I type into the filter, leave the page, then come back, the filter is active, but no value shows up in the filter input.

**Comment by TheRealJon** [comment]
Hmm, oh you know, I should probably wire the redux value to the input element value attribute ... 🤦‍♂️

**Comment by spadgett** [comment]
Yeah, I wonder also if we should preserve the filter value across page navigation. I guess it's useful when hitting the back button. It might be weird if you've switched to a different project, but the filter from the last project you viewed is still active.

**Comment by TheRealJon** [comment]
True, I can just clear it on unmount.

**Comment by TheRealJon** [comment]
@spadgett updated the overview header to reset the redux filter value when it unmounts.

**Pull request merged**